server/db/DBManager.js

```diff
@@ -1,41 +1,95 @@
+  APP_COLLECTION = APP_COLLECTION;
+
+  /**
+   * The connection string used or null if a Db object was used for the connection or the
+   * connection has not been established yet.
+   */
+  connectionString = null;
+
   constructor() {
     this.db = null;
   }
 
-  init() {
-    return new Promise((resolve, reject) => {
-      MongoClient.connect(mongoURL, { useNewUrlParser: true, useUnifiedTopology: true }, (err, client) => {
-        if (err) {
-          reject(err);
-        } else {
-          let db = client.db('opendatacam');
-          this.db = db;
-
-          // Get the collection
-          const recordingCollection = db.collection(RECORDING_COLLECTION);
-          // Create the index
-          recordingCollection.createIndex({ dateStart: -1 });
-
-          const trackerCollection = db.collection(TRACKER_COLLECTION);
-          // Create the index
-          trackerCollection.createIndex({ recordingId: 1 });
-
-          resolve(db);
-        }
+  /**
+   * Connect to the opendatacam database the MongoDB Server
+   *
+   * If connectionStringOrDbObject is a
+   *
+   * - Db object: the object pointing to a database will be used and no new connection will be
+   *   created
+   * - String: The string will be used to create a new connection to the database and then the
+   *   "opendatacam" database will be used
+   *
+   * @param {*} connectionStringOrDbObject The connection to use or credentials to create one
+   *
+   * @returns A promise that if resolved returns the opendatacam database object
+   *
+   * @throws Error if something else then a String or Db is passed
+   */
+  async connect(connectionStringOrDbObject) {
+    const createCollectionsAndIndex = function(db) {
+      const recordingCollection = db.collection(RECORDING_COLLECTION);
+      recordingCollection.createIndex({ dateStart: -1 });
+
+      const trackerCollection = db.collection(TRACKER_COLLECTION);
+      trackerCollection.createIndex({ recordingId: 1 });
+    }
+
+    const isConnectionString = typeof connectionStringOrDbObject === 'string'
+      || connectionStringOrDbObject instanceof String;
+    const isDbObject = typeof connectionStringOrDbObject === 'object';
+
+    if (isConnectionString) {
+      return new Promise((resolve, reject) => {
+        this.connectionString = connectionStringOrDbObject;
+        MongoClient.connect(this.connectionString, { useNewUrlParser: true, useUnifiedTopology: true }, (err, client) => {
+          if (err) {
+            reject(err);
+          } else {
+            let db = client.db('opendatacam');
+            this.db = db;
+
+            createCollectionsAndIndex(db);
+
+            resolve(db);
+          }
+        });
       });
-    });
+    } else if (isDbObject) {
+      this.db = connectionStringOrDbObject;
+      createCollectionsAndIndex(this.db);
+      return Promise.resolve(this.db);
+    } else {
+      return new Error();
+    }
+
+    this.connectionString = connectionStringOrDbObject;
```

Comment:
This is unreachable code. Left overs that should be removed.

Reply:
Line was removed in https://github.com/opendatacam/opendatacam/pull/311/commits/1da3a7ace61089d026c99c74ebf0df8fa550ca76